### PR TITLE
Code optimizations reported by static code analysis (2020-05-26)

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

SUMMARY: Performance "Code optimizations reported by static code analysis"

#### Purpose of change

Cleanup code while increase performance and fix minor errors.

#### Describe the solution

See commit messages.

### CODE REVIEW

**Comment by kevingranade** [comment]
Class enum is one of my favorite things in c++11 <3

File: src/activity_actor.cpp [hunk 1]
```diff
@@ -94,15 +94,15 @@ aim_activity_actor aim_activity_actor::use_bionic( const item &fake_gun,
     aim_activity_actor act = aim_activity_actor();
     act.weapon_source = WeaponSource::Bionic;
     act.bp_cost_per_shot = cost_per_shot;
-    act.fake_weapon = shared_ptr_fast<item>( new item( fake_gun ) );
+    act.fake_weapon = std::make_shared<item>( fake_gun );
```
Comment (Qrox) [inline src/activity_actor.cpp:97]:
```suggestion
    act.fake_weapon = make_shared_fast<item>( fake_gun );
```
And the same for other changes in the last commit.

Comment (ZhilkinSerg) [inline src/activity_actor.cpp:97]:
```suggestion
    act.fake_weapon = make_shared_fast<item>( fake_gun );
```

File: src/activity_actor.cpp [hunk 2]
```diff
@@ -94,15 +94,15 @@ aim_activity_actor aim_activity_actor::use_bionic( const item &fake_gun,
...
     return act;
 }
 
 aim_activity_actor aim_activity_actor::use_mutation( const item &fake_gun )
 {
     aim_activity_actor act = aim_activity_actor();
     act.weapon_source = WeaponSource::Mutation;
-    act.fake_weapon = shared_ptr_fast<item>( new item( fake_gun ) );
+    act.fake_weapon = std::make_shared<item>( fake_gun );
```
Comment (ZhilkinSerg) [inline src/activity_actor.cpp:105]:
```suggestion
    act.fake_weapon = make_shared_fast<item>( fake_gun );
```

File: src/activity_actor.cpp [hunk 3]
```diff
@@ -221,7 +221,7 @@ std::unique_ptr<activity_actor> aim_activity_actor::deserialize( JsonIn &jsin )
 
     data.read( "weapon_source", actor.weapon_source );
     if( actor.weapon_source == WeaponSource::Bionic || actor.weapon_source == WeaponSource::Mutation ) {
-        actor.fake_weapon = shared_ptr_fast<item>( new item() );
+        actor.fake_weapon = std::make_shared<item>();
```
Comment (ZhilkinSerg) [inline src/activity_actor.cpp:224]:
```suggestion
        actor.fake_weapon = make_shared_fast<item>();
```

**Pull request merged**